client/my-sites/site-settings/publishing-tools/post-by-email.tsx

```diff
@@ -0,0 +1,168 @@
+		isJetpackSiteInDevelopmentMode( state, selectedSiteId )
+	);
+	const moduleUnavailableInDevMode = useSelector( ( state ) =>
+		isJetpackModuleUnavailableInDevelopmentMode( state, selectedSiteId, moduleSlug )
+	);
+	const moduleUnavailable = siteInDevMode && moduleUnavailableInDevMode;
+	const { data: simpleSitePostByEmailSettings } = useGetPostByEmail( selectedSiteId );
+	const { mutate: simpleSiteTogglePostByEmail, isPending: isSimpleSitePendingToggle } =
+		useTogglePostByEmailMutation( selectedSiteId );
+	const { mutate: simpleSiteRegeneratePostByEmail, isPending: isSimpleSitePendingRegenerate } =
+		useRegeneratePostByEmailMutation( selectedSiteId );
+
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const handleToggleForSimpleSite = ( checked: boolean ) => {
+		if ( ! selectedSiteId ) {
+			return;
+		}
+
+		simpleSiteTogglePostByEmail( checked, {
+			onSuccess: () => {
+				dispatch( successNotice( translate( 'Settings saved successfully!' ), noticeConfig ) );
+			},
+			onError: () => {
+				dispatch(
+					errorNotice( translate( 'There was a problem saving your changes.' ), noticeConfig )
+				);
+			},
+		} );
+	};
+
+	const handleRegenerate = () => {
+		if ( ! selectedSiteId ) {
+			return;
+		}
+
+		if ( siteIsJetpack ) {
+			dispatch( jetpackRegeneratePostByEmail( selectedSiteId ) );
+			return;
+		}
+
+		simpleSiteRegeneratePostByEmail( undefined, {
+			onSuccess: () => {
+				dispatch( successNotice( translate( 'Address regenerated successfully!' ), noticeConfig ) );
+			},
+			onError: () => {
+				dispatch(
+					errorNotice( translate( 'There was a problem regenerating the address.' ), noticeConfig )
+				);
+			},
+		} );
+	};
+
+	let email;
+	if ( siteIsJetpack ) {
+		email = address && address !== 'regenerate' ? address : '';
+	} else {
+		email = simpleSitePostByEmailSettings?.email;
+	}
+	const isActive = siteIsJetpack
+		? jetpackPostByEmailIsActive
+		: simpleSitePostByEmailSettings?.isEnabled;
+	const isDisabledForBothFlows =
+		isFormPending ||
+		jetpackRegeneratingPostByEmail ||
+		isSimpleSitePendingRegenerate ||
+		! isActive ||
+		!! moduleUnavailable;
+
+	return (
+		<>
+			<FormFieldset>
+				<SupportInfo
+					text={ translate(
+						'Allows you to publish new posts by sending an email to a special address.'
+					) }
+					link={
+						siteIsAtomic
+							? localizeUrl( 'https://wordpress.com/support/post-by-email/' )
+							: 'https://jetpack.com/support/post-by-email/'
+					}
+					privacyLink={ ! siteIsAtomic }
+				/>
+				{ siteIsJetpack ? (
+					<JetpackModuleToggle
+						siteId={ selectedSiteId }
+						moduleSlug="post-by-email"
+						label={ translate( 'Publish posts by sending an email' ) }
+						disabled={ isFormPending || moduleUnavailable }
+					/>
+				) : (
+					<ToggleControl
+						checked={ !! simpleSitePostByEmailSettings?.isEnabled }
+						disabled={ isFormPending || isSimpleSitePendingToggle || isSimpleSitePendingRegenerate }
```

Comment:
```suggestion
	const isJetpackDisabled = isFormPending || moduleUnavailable;
	const isSimpleSiteDisabled =
		isFormPending || isSimpleSitePendingToggle || isSimpleSitePendingRegenerate;

	const isDisabledControls = ! isActive || isJetpackDisabled || isSimpleSiteDisabled;

	return (
		<>
			<FormFieldset>
				<SupportInfo
					text={ translate(
						'Allows you to publish new posts by sending an email to a special address.'
					) }
					link={
						siteIsAtomic
							? localizeUrl( 'https://wordpress.com/support/post-by-email/' )
							: 'https://jetpack.com/support/post-by-email/'
					}
					privacyLink={ ! siteIsAtomic }
				/>

				{ siteIsJetpack ? (
					<JetpackModuleToggle
						siteId={ selectedSiteId }
						moduleSlug="post-by-email"
						label={ translate( 'Publish posts by sending an email' ) }
						disabled={ isJetpackDisabled }
					/>
				) : (
					<ToggleControl
						checked={ !! simpleSitePostByEmailSettings?.isEnabled }
						disabled={ isSimpleSiteDisabled }
```
It'd be nice to assign the `disabled` values for the respective controls to separate variables.

Reply:
Oh yes, that's much cleaner, thank you!